python/ray/_private/runtime_env.py

```diff
@@ -141,7 +141,7 @@ def _hash_modules(path: Path) -> bytes:
             with open(from_file_name, mode="rb") as f:
                 data = f.read(BUF_SIZE)
```

Comment:
Hmm @iycheng isn't this only including the first BUF_SIZE bytes from each file then?

Reply:
Yes, I feel I forgot the loop here.  It should be:

```
data = f.read(BUF_SIZE)
while data is not None:
    md5.update(data)
    data = f.read(BUF_SIZE)
```

@valiantljk-tt thanks for the fix. Could you update it to something like this?

Reply:
sure

Reply:
one question, do we need to read all data from the file to produce the hash? what if the file is very large?
I thought, the purpose of this is generate a unique name, so first BUF_SIZE bytes from each file should be fine, right?
@iycheng

Reply:
We need a better algorithm for this one in the future. Right now, to make it safe, we calculate caching for all files. Later, we can cache the hash value of these files locally and only recalculate them when the data changed.

Reply:
ok, that makes sense, let me submit the change.

Reply:
@iycheng just submitted, plz review. thx